### PR TITLE
Adding image pull secrets to job, cronjob kinds

### PR DESCRIPTION
**Pull request by ds-ms**
Fixes https://github.com/microsoft/azure-pipelines-tasks/issues/10580

### CODE REVIEW

File: Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts [hunk 1]
```diff
@@ -239,7 +231,11 @@ function getImagePullSecrets(inputObject: any) {
         return null;
     }
 
-    if (isEqual(inputObject.kind, KubernetesWorkload.Pod, StringComparer.OrdinalIgnoreCase)) {
+    if (isEqual(inputObject.kind, KubernetesWorkload.cronjob, StringComparer.OrdinalIgnoreCase)) {
+        return inputObject.spec.jobTemplate.spec.template.spec.imagePullSecrets;
```
Comment (nishubansal) [inline Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts]: null check for jobTemplate required? Also where is the handling for Workload type - job ?

File: Tasks/KubernetesManifestV0/Tests/L0.ts [hunk 2]
```diff
@@ -250,4 +251,22 @@ describe('Kubernetes Manifests Suite', function () {
...
+        const cronJobObject = yaml.load(cronJobFile);
+        updateImagePullSecrets(cronJobObject, ['privaterepo-secret'], true);
+        assert(cronJobObject.spec.jobTemplate.spec.template.spec.imagePullSecrets[0].name === 'privaterepo-secret', 'should have updated the image pull secret correctly');
+        done();
+    });
+
+    it('Run should successfully add image pull secrets to a job', (done: MochaDone) => {
+        const testFile = path.join(__dirname, './manifests/', 'job.yaml');
+        const cronJobFile = fs.readFileSync(testFile).toString();
+        const cronJobObject = yaml.load(cronJobFile);
```
Comment (nishubansal) [inline Tasks/KubernetesManifestV0/Tests/L0.ts]: change the variable name to jobObject and jobFile

File: Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts (hunk 1 above)
Comment (ds-ms) [reply Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts]: Job has a default spec template. It gets handled in the default workload `if..else` case.

Comment (ds-ms) [reply Tasks/KubernetesManifestV0/src/utils/KubernetesObjectUtility.ts]: Surrounded the fetching and overriding logic with a `try..catch` block

**Pull request merged**